test/blackbox-tests/test-cases/invalid-package-version.t

```diff
@@ -19,7 +19,7 @@ Building doesn't work as the dependency is invalid.
   Error: "foo.1.2.3" is an invalid opam package name.
   Package names can contain letters, numbers, '-', '_' and '+', and need to
   contain at least a letter.
-  Hint: foo_1_2_3 would be a correct opam package name
+  Hint: (foo (= 1.2.3)) would be a correct opam package name
```

Comment:
This is no longer an "opam package name" but a "package dependency".

Reply:
Any other time we validate a package name would also give a hint like `(foo (= 1.2.3))` even if that syntax is not accepted.

Reply:
This is a good point, it should be constrained to places where we can potentially generate opam files from.

Reply:
I believe my last commit solves all of these comments.